demos/python_demos/common/images_capture.py

```diff
@@ -0,0 +1,156 @@
+import os
+import sys
+import copy
+
+import cv2
+
+
+class InvalidInput(Exception):
+
+    def __init__(self, message):
+        self.message = message
+
+
+class OpenError(Exception):
+
+    def __init__(self, message):
+        self.message = message
+
+
+class ImagesCapture:
+
+    def __init__(self, loop):
+        self.loop = loop
+
+    def read():
+        raise NotImplementedError
+
+    def fps():
+        raise NotImplementedError
+
+    def get_type():
+        raise NotImplementedError
+
+
+class ImreadWrapper(ImagesCapture):
+
+    def __init__(self, input, loop):
+        super().__init__(loop)
+        if not os.path.isfile(input):
+            raise InvalidInput("Can't find the image by {}".format(input))
+        self.image = cv2.imread(input, cv2.IMREAD_COLOR)
+        if self.image is None:
+            raise OpenError("Can't open the image from {}".format(input))
+        self.can_read = True
+
+    def read(self):
+        if self.loop:
+            return copy.deepcopy(self.image)
+        if self.can_read:
+            self.can_read = False
+            return copy.deepcopy(self.image)
+        return None
+
+    def fps(self):
+        return 1.0
+
+    def get_type(self):
+        return 'IMAGE'
+
+
+class DirReader(ImagesCapture):
+
+    def __init__(self, input, loop):
+        super().__init__(loop)
+        self.dir = input
+        if not os.path.isdir(self.dir):
+            raise InvalidInput("Can't find the dir by {}".format(input))
+        self.names = sorted(os.listdir(self.dir))
+        if not self.names:
+            raise OpenError("The dir {} is empty".format(input))
+        self.file_id = 0
+        for name in self.names:
+            filename = os.path.join(self.dir, name)
+            image = cv2.imread(filename, cv2.IMREAD_COLOR)
+            if image is not None:
+                return
+        raise OpenError("Can't read the first image from {}".format(input))
+
+    def read(self):
+        while self.file_id < len(self.names):
+            filename = os.path.join(self.dir, self.names[self.file_id])
+            image = cv2.imread(filename, cv2.IMREAD_COLOR)
+            self.file_id += 1
+            if image is not None:
+                return image
+        if self.loop:
+            self.file_id = 0
+            while self.file_id < len(self.names):
+                filename = os.path.join(self.dir, self.names[self.file_id])
+                image = cv2.imread(filename, cv2.IMREAD_COLOR)
+                self.file_id += 1
+                if image is not None:
+                    return image
+        return None
+
+    def fps(self):
+        return 1.0
+
+    def get_type(self):
+        return 'DIR'
+
+
+class VideoCapWrapper(ImagesCapture):
```

Comment:
Please split this class into 2 separate classes. At next step we will implement check for stream data depletion (as separate PR I presume), so we'll definetely have to doit  in separate class for video reader. So, please split this class into VideoFileReader and CameraReader clsasses.

Reply:
Wouldn't it make sense to wait until there's actually a difference between the two cases before splitting the class?

Reply:
It's better make consistant sloution from the beginning - other cases follow one case -one class paradigm. Well, if it really would take much time to implement, I'd agree to postpone this, but I believe this is not the case.

Reply:
It is easy to implement, but the difference between two classes will be only in `__init__`, so the `self.read()` in class `CameraReader` looks a bit redundant. Also, we can check stream data depletion (when the `self.read()` returns None) only if the class `VideoCapWrapper` has a video type, otherwise raise some exception about interruption of camera capture.

Reply:
Well, init is 1/2 of whole class code - so it's quite enough. Also, there're questions about read function - do we really need to self.cap.set(cv2.CAP_PROP_POS_FRAMES, 0) for camera ? loop won't work there, so it could be erased too. So, there would be quite different code, particulalry after implementing check for the end of stream

Reply:
I agree with you about camera's loop, so I will split it into two classes.